### DISCUSSION TITLE
Build failure on MacOSx (M2) Sonoma

### DISCUSSION THREAD

**Discussion by DeniseWorthen** [JCSDA/spack-stack#1002] [General]
I didn't want to create an issue, because most likely this is user-error. I decided to see if I could build spack-stack (ufs-weather-model environment) on my home platform. My platform is new and very clean. I did a selective migration from my old Intel machine. 

I started at [6.1](https://spack-stack.readthedocs.io/en/latest/NewSiteConfigs.html#newsiteconfigs). I was able to follow the instructions through to the Section 10. I did not do anything under Section 9. 

For the concretize, I see some items marked with [e]. I don't know if that indicates an error, but I got through that step. At the next step (install), it failed with the last thing in the build-out showing

```
checking for on_exit... no
configure: WARNING: neither statfs() and statvfs() were found
configure: error: Cannot continue
```

[spack-build-out.txt](https://github.com/JCSDA/spack-stack/files/14232865/spack-build-out.txt)
[log.concretize.txt](https://github.com/JCSDA/spack-stack/files/14232868/log.concretize.txt)

**Comment by climbfuji** [comment]
You may be hitting this: https://github.com/JCSDA/spack-stack/issues/971

**Comment by DeniseWorthen** [comment]
@climbfuji Yes, thanks, that certainly seems related. It's difficult for me to follow what has been merged or still needs to be merged. There is a comment that  jscda/spack-stack/977 has been merged, and that it may fix the issue. I have that commit in what I was trying to build (b7b36e0).

**Reply by climbfuji** [reply]
Can you try to set these flags globally in your spack compiler config?
```
LDFLAGS="-Wl,-ld_classic"
```
See https://spack.readthedocs.io/en/latest/getting_started.html#compiler-config

**Reply by srherbener** [reply]
I agree to try setting LDFLAGS as @climbfuji suggests.

If that doesn't work, @fabiolrdiniz got it to work on his Mac with this edit to the openmpi/package.py script: https://github.com/JCSDA/spack-stack/issues/971#issuecomment-1928091205

**Comment by DeniseWorthen** [comment]
Add the ``ldflags`` (I manually edited the site/compilers.yaml) as suggested by @climbfuji worked. At least the the ``install`` reports success, I haven't tried to use it yet.

One thing I found is that it did not seem to work to try to "rebuild" in my previous env, so I created a new one. Is there a "right" way to clean an env ?

**Reply by climbfuji** [reply]
I think you have to force a re-concretization for the new flags to be effective
